js/effects/sparkle.js

```diff
@@ -0,0 +1,131 @@
+import Effect, { ConfigUI, fract } from './effect';
+import { parseHtml } from '../ui/util';
+
+const EffectName = 'Sparkle';
+const EffectDescription = 'Particle size and brightness increase randomly';
+
+class SparkleConfigUI extends ConfigUI {
+  constructor() {
+    super();
+    this.element = parseHtml(`
+      <fieldset>
+        <legend>${EffectName}</legend>
+      </fieldset>
+    `);
+    const ui = this.element;
+  }
+
+  getElement() {
+    return this.element;
+  }
+
+  getConfig() {
+    const config = {};
+
+    return config;
+  }
+
+  applyConfig(config) {
+  }
+}
+
+export default class SparkleEffect extends Effect {
+  static register(instance, props, uniforms, vertexShader, frag, attributes) {
+    // Params
+    const scaleMin = 1;
+    const scaleMax = 2;
+    const sparklePercentage = 0.7;
+    const sparkleTime = Math.round(instance.getPeriod() / 20);
+
+    // The next line is supposed to calculate the likelyhood by which
+    // particles have to start to sparkle if we want to have an average 
+    // of <sparklePercentage> of all particles sparkling within 100 frames
+    // FIXME the 100 frames are pretty arbitrary
+    const prob = Math.pow(1 - sparklePercentage, 1 / 100);
+    const shouldStart = () => {
+      if (props.clock.getTime() + sparkleTime > instance.timeEnd)
+        return false;
+      if (Math.random() < prob)
+        return false;
+      return true;
+    };
+
+    // Shader values
+    const particlesCount = props.config.xParticlesCount * props.config.yParticlesCount;
+    const progresses = new Float32Array(particlesCount);
+    const { id: bufId, buffer } = props.state.createBuffer({
+      usage: 'dynamic',
+      type: 'float',
+      length: particlesCount,
+      data: progresses
+    });
+    
+    const progress = attributes.add('progress', 'float', (ctx, props) => {
+      const time = props.clock.getTime();
+      if (instance.timeBegin <= time && time <= instance.timeEnd) {
+        for (let i = 0; i < progresses.length; i++) {
+          if (progresses[i] != 0 || shouldStart()) {
+            progresses[i] += props.clock.getDelta() / sparkleTime;
+            if (progresses[i] > 1) {
+              progresses[i] = 0;
+            }
+          }
+        }
+        buffer({ data: progresses });
```

Comment:
Looping over the data and changing it for each frame seems possibly a bit inefficient - not sure. If you have a better idea, I'm all open for it